### ISSUE TITLE
is the published geneformer for census 2024-07-01 fine-tuned

### ISSUE THREAD

**Issue by vtrifonov-altos** [chanzuckerberg/cellxgene-census#1355]
I am not sure if this is a bug or not but I am looking at the "fine-tuned" Geneformer model published at this link [s3://cellxgene-contrib-public/models/geneformer/2024-07-01/homo_sapiens/fined-tuned-model](url) and it does not look like it was actually fine-tuned. There are a few things that make me suspect that this link has a pre-trained model without additional fine-turning for cell classification:
1. when loading the model I get a warning: Some weights of BertForSequenceClassification were not initialized from the model checkpoint and are newly initialized: ['bert.pooler.dense.bias', 'bert.pooler.dense.weight', 'classifier.bias', 'classifier.weight']
2. the loaded model has only 2 classes for the classifier task-head (the Geneformer 2023-12-15 census version has 88)
3. the [s3://cellxgene-contrib-public/models/geneformer/2024-07-01/homo_sapiens/fined-tuned-model/config.json](url) does not contain any information for the cell-classification training. For example [s3://cellxgene-contrib-public/models/geneformer/2023-12-15/homo_sapiens/fined-tuned-model/config.json](url)  shows 88 labels used for the classification

```
aws s3 --no-sign-request cp s3://cellxgene-contrib-public/models/geneformer/2024-07-01/homo_sapiens/fined-tuned-model/config.json - 
{
  "architectures": [
    "BertForMaskedLM"
  ],
  "attention_probs_dropout_prob": 0.02,
  "classifier_dropout": null,
  "hidden_act": "relu",
  "hidden_dropout_prob": 0.02,
  "hidden_size": 512,
  "initializer_range": 0.02,
  "intermediate_size": 1024,
  "layer_norm_eps": 1e-12,
  "max_position_embeddings": 4096,
  "model_type": "bert",
  "num_attention_heads": 8,
  "num_hidden_layers": 12,
  "pad_token_id": 0,
  "position_embedding_type": "absolute",
  "torch_dtype": "float32",
  "transformers_version": "4.37.2",
  "type_vocab_size": 2,
  "use_cache": true,
  "vocab_size": 20275
}
```

<img width="1461" alt="Image" src="https://github.com/user-attachments/assets/a5b82521-bf41-4dd9-8b72-d65c58030c6a" />

**Comment by mlin** [comment]
@vtrifonov-altos The released Geneformer embeddings for 2024-07 were generated using a not-then-quite-final version of the  [gf-12L-95M-i4096_MTLCellClassifier_CELLxGENE_240522](https://huggingface.co/ctheodoris/Geneformer/tree/main/fine_tuned_models/gf-12L-95M-i4096_MTLCellClassifier_CELLxGENE_240522) model fine-tuned by the Geneformer authors themselves. It was shortly before they released that model, and regrettably I'm unsure what delta there may have been before the final release, but I can seek to clarify that with the authors if needed. The embeddings for the new 2025-01-30 Census were generated using the released model I linked.

I'll need to do some archaeology on what was uploaded to `s3://cellxgene-contrib-public/models/geneformer/2024-07-01/homo_sapiens/fined-tuned-model/` which I didn't do personally. Without having yet looked into it, I'd say there is a high probability it is NOT the pre-release model actually used, since it was not supposed to be released at that time IIRC, and we'd generated multiple candidate sets of embeddings using different approaches -- a mixup/oversight could have occurred easily. Sorry about this and thanks for bringing it to our attention.

**Comment by vtrifonov-altos** [comment]
Thanks @mlin! I checked and the checkpoint that was published `s3://cellxgene-contrib-public/models/geneformer/2024-07-01/homo_sapiens/fined-tuned-model/` is the same as [gf-12L-95M-i4096_MTLCellClassifier_CELLxGENE_240522](https://huggingface.co/ctheodoris/Geneformer/tree/main/fine_tuned_models/gf-12L-95M-i4096_MTLCellClassifier_CELLxGENE_240522). I also verified with Geneformer and they told me that this checkpoint although it was fine-tuned for classification only has the trunk of the model.
